cpp/oneapi/dal/algo/basic_statistics/backend/cpu/finalize_compute_kernel_dense.cpp

```diff
@@ -50,11 +50,23 @@ static compute_result<Task> call_daal_kernel_finalize_compute(
     const context_cpu& ctx,
     const descriptor_t& desc,
     const partial_compute_result<Task>& input) {
-    const auto result_ids = daal_lom::estimatesAll;
+    const auto result_ids = get_daal_estimates_to_compute(desc);
     const auto daal_parameter = daal_lom::Parameter(result_ids);
 
-    auto column_count = input.get_partial_min().get_column_count();
+    const auto res_min_max = result_options::min | result_options::max;
+    const auto res_mean_varc = result_options::mean | result_options::variance;
+    std::int64_t column_count = 10;
 
+    const auto res_op = desc.get_result_options();
+
+    if (res_op.test(res_mean_varc)) {
+        column_count = input.get_partial_sum().get_column_count();
+    }
+    if (res_op.test(res_min_max)) {
+        column_count = input.get_partial_min().get_column_count();
+    }
```

Comment:
Maybe I do not understand something. But this is a very suspicious change. What if the user does not compute any of {min, max, mean, variance}, and computes, for example, 'sum_squares' only. Then column_count will remain == 10. Is it Ok?

Reply:
Also the use of a magic constant '10' is not a good practice anyway. Better change to zero + adding the assert after the actual column_count is defined in the code .

Reply:
Good point! Will fix in next commit!